samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.runtime;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.config.TaskConfig;
+import org.apache.samza.container.RunLoop;
+import org.apache.samza.container.SamzaContainer;
+import org.apache.samza.processor.StreamProcessorTestUtils;
+import org.apache.samza.task.StreamTask;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.*;
+
+
+public class TestClusterBasedProcessorLifecycleListener {
+
+  private SamzaContainer container;
+  private RunLoop mockRunLoop;
+  private ClusterBasedProcessorLifecycleListener clusterBasedProcessorLifecycleListener;
+  private ProcessorLifecycleListener processorLifecycleListener;
+  private Runnable mockOnSignal;
+
+  @Before
+  public void setup() {
+    mockRunLoop = mock(RunLoop.class);
+    mockOnSignal = mock(Runnable.class);
+    container = StreamProcessorTestUtils.getDummyContainer(mockRunLoop, Mockito.mock(StreamTask.class));
```

Comment:
A unit test for `ClusterBasedProcessorLifecycleListener` should not require setting up a `SamzaContainer`. You just need to call the methods of the listener directly.
If you would like to test that the listener lifecycle methods are properly called, then those tests belong in the test for `SamzaContainer`.

Reply:
Good point! Seems like the tests for `SamzaContainer` does this check already. Fixed my tests.